### PR TITLE
docs: move all bazel testing info to a single location

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
instead of presenting the same (or similar information) in both the
DEVELOPER.md and the BAZEL.md files, more all the information in the
BAZEL file and refer to that section in the DEVELOPER file

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Issue

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
 - basically I've been referring to the BAZEL file, but stumbled across the explanation of the `...` in the DEVELOPER one, which seemed quite valuable for the BAZEL file as well, instead of copy that section I just figured that the DEVELOPER's md should just point to the BAZEL one and that one should be the only point of reference (as it also contains various information for debugging)

### CODE REVIEW

File: docs/BAZEL.md [hunk 1]
```diff
@@ -48,9 +48,18 @@ new as of May 2017 and not very stable yet.
 
 ## Testing Angular
 
-- Test package in node: `yarn bazel test packages/core/test:test`
-- Test package in karma: `yarn bazel test packages/core/test:test_web`
-- Test all packages: `yarn bazel test packages/...`
+- Test package in node: `yarn test packages/core/test:test`
+- Test package in karma: `yarn test packages/core/test:test_web`
+- Test all packages: `yarn test packages/...`
```
Comment (dario-piotrowicz) [inline docs/BAZEL.md:53]: I think that `bazel` is not necessary

actually the script seem to use `bazelisk` (I guess they are the same-ish?):
Screenshot at 2022-05-21 18-20-13

File: docs/DEVELOPER.md [hunk 2]
```diff
@@ -74,24 +74,15 @@ node ./scripts/build/build-packages-dist.js
 
 ## Running Tests Locally
 
-Bazel is used as the primary tool for building and testing Angular. Building and testing are
-incremental with Bazel, and it's possible to only run tests for an individual package instead
-of for all packages. Read more about this in the [BAZEL.md](./BAZEL.md) document.
```
Comment (dario-piotrowicz) [inline docs/DEVELOPER.md:79]: I don't think that explaining why bazel is used adds much here :thinking:

File: docs/DEVELOPER.md [hunk 3]
```diff
@@ -74,24 +74,15 @@ node ./scripts/build/build-packages-dist.js
...
-**Note**: The ellipsis in the commands above is not meant to be substituted by a package name, but
-is used by Bazel as a wildcard to execute all tests in the specified path. To execute tests for a
-single package, the commands are (exemplary):
-- `yarn test //packages/core/...` for all tests, or
-- `yarn test //packages/core/test:test_web_firefox` for a particular test suite.
+Note that you should execute all test suites before submitting a PR to GitHub (`yarn test //packages/...`).
 
-**Note**: The first test run will be much slower than future runs. This is because future runs will
-benefit from Bazel's capability to do incremental builds.
+However the tests will be executed on our Continuous Integration infrastructure. So if you forgot to run the tests and some broke github will present you with the CI errors which you can then fix.
```
Comment (dario-piotrowicz) [inline docs/DEVELOPER.md:83]: Honestly I have never run locally all the tests, on my (dinosaur) machine it would take ages, so I think it could be valuable to suggest to do that, but provide the alternative to rely on the CI

Comment (devversion) [reply docs/DEVELOPER.md:83]: Technically if you run everything once locally, you would get the same caching benefits as CI, but in practice this is obviously still slower. Just want to avoid recommend to rely fully on CI. Ideally folks would run some tests for the parts they have changed. e.g. if I change the compiler, I run tests in the compiler/test Bazel package.

Comment (devversion) [inline docs/DEVELOPER.md:83]:
```suggestion
However, affected tests will be executed on our CI infrastructure. So if you forgot to run some affected tests which would fail, GitHub will indicate the error state and present you the failures.
```

**Review by devversion — APPROVED** [review]
LGTM overall

File: docs/DEVELOPER.md (hunk 3 above)
Comment (dario-piotrowicz) [reply docs/DEVELOPER.md]: Yeah the first run is definitely my issue, after that the tests do run rather quick :slightly_smiling_face: 

anyways I hope that the wording here is fine then, sort of letting developers know that CI checks will be performed but not actively recommending to rely solely on that :slightly_smiling_face:

**Comment by AndrewKushnir** [comment]
Merge-assistance: the `legacy-unit-tests-saucelabs` CI failure is unrelated to the changes in this PR.

**Comment by alxhub** [comment]
This PR was merged into the repository by commit f04a8ead1869264e0ffafecdcbfa3c6bad14bf06.
